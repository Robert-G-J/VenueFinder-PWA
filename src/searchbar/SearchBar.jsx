import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";

const SearchBar = ({ updateSearchbarQuery, getVenues }) => {
  const debounceUpdateSearchbarQuery = _.debounce(updateSearchbarQuery, 250);

  return (
    <div className="search-bar">
      <label htmlFor="search-bar" className="search-bar__label">
        Search for spots
      </label>
      <input
        type="text"
        id="search-bar"
        className="search-bar__input"
        placeholder="Search spots..."
        onChange={event => debounceUpdateSearchbarQuery(event.target.value)}
      />
      <button className="search-bar__button" onClick={getVenues}>
        Search near me
      </button>
    </div>
  );
};

SearchBar.propTypes = {
  getVenues: PropTypes.func.isRequired,
  updateSearchbarQuery: PropTypes.func.isRequired
};

export default SearchBar;
