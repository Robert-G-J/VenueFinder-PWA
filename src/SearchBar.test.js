import React from "react";
import SearchBar from "./SearchBar";
import renderer from "react-test-renderer";

describe("The search bar componenet", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
