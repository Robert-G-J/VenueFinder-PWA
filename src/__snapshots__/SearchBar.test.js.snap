// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The search bar componenet renders correctly 1`] = `
<div
  className="search-bar"
>
  <label
    className="search-bar__label"
    htmlFor="search-bar"
  />
  <input
    className="search-bar__input"
    onChange={undefined}
    placeholder="Search spots..."
    type="text"
  />
  <button
    className="search-bar__button"
    onClick={undefined}
  >
    Search
  </button>
</div>
`;
